client/state/billing-data/util.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+import { moment } from 'i18n-calypso';
+
+/**
+ * Parses the date within a transaction.
+ * Returns a copy of the updated transaction.
+ *
+ * @param  {Object}  transaction  Transaction object
+ * @return {Object}               Updated transaction with parsed date
+ */
+export const parseDate = ( transaction ) => {
+	return assign( {}, transaction, {
```

Comment:
> return { ...transaction, date: moment( transaction.date ).toDate() };

Looks much simpler this way!

> Also is Date.parse too unreliable for our date string?

Turns out it's reliable enough. I've updated it to use native `Date` instead of `moment`.